api/v1alpha1/release_types.go

```diff
@@ -196,23 +201,29 @@ func (r *Release) MarkSucceeded() {
 	}
 
 	r.Status.CompletionTime = &metav1.Time{Time: time.Now()}
-	r.setStatusCondition(metav1.ConditionTrue, ReleaseReasonSucceeded)
+	r.setStatusCondition(releaseConditionType, metav1.ConditionTrue, ReleaseReasonSucceeded)
 
 	go metrics.RegisterCompletedRelease(ReleaseReasonSucceeded.String(), r.Status.ReleaseStrategy, r.Status.Target,
 		r.Status.StartTime, r.Status.CompletionTime, false)
 }
 
-// SetCondition creates a new condition with the given status and reason. Then, it sets this new condition,
+// RegisterDeploymentStatus registers the AllComponentsDeployed status from the SnapshotEnvironmentBinding to the release.
+func (r *Release) RegisterDeploymentStatus(binding *applicationapiv1alpha1.SnapshotEnvironmentBinding) {
```

Comment:
The Release should know nothing about the binding. This function should be in the adapter and use Mark functions in here to set the status (eg. MarkInvalid, MarkRunning...). Good names for those would be MarkDeploying and MarkDeployed.

The function doing the same work for the Release PipelineRun is `registerReleasePipelineRunStatus` in the adapter.